### PR TITLE
External video widget

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I tested the new widget manually and added automated tests.

#### Why is this the best possible solution? Were any other approaches considered?
It's a continuation of: https://github.com/getodk/collect/issues/4214

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
This time I didn't touch other widgets so we can focus on the new External video widget only and that would be enough.

#### Do we need any specific form for testing your changes? If so, please attach one.
[exTestForm.xlsx](https://github.com/getodk/collect/files/5919103/exTestForm.xlsx)

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/widgets/ExVideoWidget.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            answerFile = (File) object;
+            if (answerFile.exists()) {
+                questionMediaManager.replaceAnswerFile(getFormEntryPrompt().getIndex().toString(), answerFile.getAbsolutePath());
+                binding.playVideoButton.setEnabled(true);
+                widgetValueChanged();
+            } else {
+                Timber.e("Inserting Video file FAILED");
+            }
+        } else if (object != null) {
+            Timber.e("ExVideoWidget's setBinaryData must receive a video file but received: %s", object.getClass());
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/widgets/ExVideoWidget.java]: I think if we do want to log something here we should log the MIME type. I don't really know if that's actionable on our end, though. It probably just means that a form called out to an app that can return files other than videos.

I don't think the user gets any feedback, though? Could we do a toast that says something like "Application returned an invalid file type"?

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/widgets/ExVideoWidget.java]: Ok I added it.

**Review by lognaturel — APPROVED** [review]

**Review by lognaturel — APPROVED** [review]

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1, 7.1 and 9.0.

Tested cases:
generated answer
RTL language
used forward/backward navigation buttons
remove response
save/finalize form
toast is visible when an external app is not available
launch counter app
light/dark mode
edit form
don't keep activities
rotation
minimize
send a submission to Central and Aggregate
different font sizes
regression check on ex string and numerical widgets

**Comment by mmarciniak90** [comment]
Tested also with success on Android 7.0, 8.1, 10.0

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Comment by grzesiek2010** [comment]
Closed in favor of https://github.com/getodk/collect/pull/4382

**Pull request closed without merging**